.circleci/config.yml

```diff
@@ -38,31 +41,81 @@ jobs:
       - store_artifacts:
           path: build/reports/jacoco
   integration-test:
-    executor:
-      name: hmpps/localstack
-      jdk_tag: << pipeline.parameters.java-version >>
-      localstack_tag: << pipeline.parameters.localstack-version >>
-      services: "sns,sqs"
-    steps:
-      - checkout
-      - restore_cache:
-          keys:
-            - gradle-{{ checksum "build.gradle.kts" }}
-            - gradle-
-      - run:
-          command: ./gradlew -Dorg.gradle.jvmargs="--illegal-access=permit" -Dkotlin.daemon.jvm.options="--illegal-access=permit" integrationTest
-      - save_cache:
-          paths:
-            - ~/.gradle
-          key: gradle-{{ checksum "build.gradle.kts" }}
-      - store_test_results:
-          path: build/test-results
-      - store_artifacts:
-          path: build/reports/tests
+        machine: true
+        environment:
+          POSTGRES_VERSION: << pipeline.parameters.postgres-version >>
+          POSTGRES_DB: "create-and-vary-a-licence-db"
+          POSTGRES_USER: "cvl"
+          POSTGRES_PASSWORD: "cvl"
+          POSTGRES_PORT: "5433"
+          LOCALSTACK_VERSION: << pipeline.parameters.localstack-version >>
+          LOCALSTACK_SERVICES: "sqs,sns"
+        steps:
+          - checkout
+          - restore_cache:
+                keys:
+                  - gradle-cache-{{ checksum "build.gradle.kts" }}
+                  - gradle-cache-
+          - run:
+                name: Create Docker network
+                command: docker network create test-net
+          - run:
+                name: Start PostgreSQL in RAM
+                command: |
+                  docker run -d \
+                    --rm \
+                    --name postgres \
+                    --network test-net \
+                    -e POSTGRES_DB=$POSTGRES_DB \
+                    -e POSTGRES_USER=$POSTGRES_USER \
+                    -e POSTGRES_PASSWORD=$POSTGRES_PASSWORD \
+                    --tmpfs /var/lib/postgresql/data:rw,noexec,nosuid,size=512m \
```

Comment:
wonder whether pointing [this](https://circleci.com/docs/using-docker/#ram-disks) would help?